hk_adc/hk_adc.c

```diff
@@ -0,0 +1,530 @@
+/*
+ * hk_adc.c
+ *
+ *  Created on: 05-Mar-2021
+ *      Author: Sonal Shrivastava
+ */
+
+#include "../firmware/drivers/mss_gpio/mss_gpio.h"
+#include "../firmware/drivers/mss_i2c/mss_i2c.h"
+#include "hk_adc.h"
+
+
+#define TX_LENGTH                       1u
+#define RX_LENGTH                       2u
+#define DUMMY_SERIAL_ADDR               0xFF
+#define NUM_SAMPLES_TO_AVG              10u
+
+
+/* Local Variables */
+static mss_i2c_status_t status;
+static uint16_t read_value;
+static uint8_t os_bit = 0;
+static uint8_t  rx_buffer[RX_LENGTH];
+static uint16_t volt_array[NUM_SAMPLES_TO_AVG];
+static uint16_t curr_array[NUM_SAMPLES_TO_AVG];
+static uint16_t citi_temp_array[NUM_SAMPLES_TO_AVG];
+static uint16_t hk_adc_avg_volt;
+static uint16_t hk_adc_avg_curr;
+static uint16_t hk_adc_avg_citi_temp;
+
+
+/* Local Function prototypes */
+static int hk_adc_reg_write(hk_adc_register_t reg, uint8_t *write_buffer);
+static int hk_adc_reg_read(hk_adc_register_t reg, uint16_t *read_buffer);
+static void hk_adc_update_multiplier_volt(void);
```

Comment:
Function no longer exists, please remove the prototype declaration.

Reply:
Sorry, missed out on this one previously. Removed it now.